protocols/v2/binary-sv2/no-serde-sv2/codec/src/datatypes/mod.rs

```diff
@@ -1,3 +1,29 @@
+/// Provides implementations for encoding and decoding data types in the SV2 protocol,
+/// supporting both fixed-size and dynamically-sized types. Defines the `Sv2DataType` trait,
+/// which standardizes serialization and deserialization across various types, including
+/// those with custom requirements like byte padding and dynamic sizing.
+///
+/// ## Structure and Contents
+///
+/// ### `Sv2DataType` Trait
+/// The `Sv2DataType` trait offers methods to:
+/// - **Deserialize**: Convert byte slices or reader sources into Rust types.
+/// - **Serialize**: Encode Rust types into byte slices or write them to I/O streams.
+///
+/// Supports both **checked** and **unchecked** variants for serialization and deserialization.
+/// Checked functions validate data lengths, while unchecked versions assume size correctness for optimized performance.
+///
+/// ### Modules
+/// - **`copy_data_types`**: Defines fixed-size types directly copied into or from byte slices, such as `U24` (24-bit unsigned integer).
+/// - **`non_copy_data_types`**: Manages dynamically-sized types, like sequences, public keys, and strings, requiring size handling logic for SV2 compatibility.
+///
+/// ### Re-exports
+/// Re-exports common data types used in SV2 serialization, such as `PubKey`, `Signature`, `Seq0255`, and others, simplifying protocol data handling with concrete
+/// implementations of `Sv2DataType`.
+///
+/// The `Sv2DataType` trait and its implementations enable seamless conversion between in-memory representations and serialized forms, ensuring efficient protocol
+/// communication and interoperability.
+
```

Comment:
I cannot find a rendering of this header anywhere in the Rust Docs when I do `cargo doc`

no point in using `///` here